packages/build/src/tarball.ts

```diff
@@ -322,52 +330,40 @@ export type TarballFile = { path: string; contentType: string };
 /**
  * Create a gzipped tarball or zip for the provided options.
  *
- * @param {string} input - The file location to tarball.
- * @param {string} outputDir - Where to save the tarball.
- * @param {string} buildVariant- The build variant.
- * @param {string} version - The version.
- *
- * @returns {TarballFile} The path and type of the tarball.
+ * @param outputDir - Where to save the tarball.
+ * @param buildVariant - The build variant.
+ * @param packageInformation - Additional package information.
+ * @returns The path and type of the tarball.
  */
 export async function createTarball(
   outputDir: string,
-  buildVariant: string,
+  buildVariant: BuildVariant,
   packageInformation: PackageInformation
 ): Promise<TarballFile> {
-  const tarballFilename = tarballPath(
-    outputDir,
-    buildVariant,
-    packageInformation.metadata.version,
-    packageInformation.metadata.name);
-
-  console.info('mongosh: gzipping:', tarballFilename);
-
-  if (buildVariant === BuildVariant.Linux) {
-    await tarballPosix(packageInformation, tarballFilename);
-
-    return {
-      path: tarballFilename,
-      contentType: 'application/gzip'
-    };
-  } else if (buildVariant === BuildVariant.Redhat) {
-    await tarballRedhat(packageInformation, packageInformation.rpmTemplateDir, tarballFilename);
-
-    return {
-      path: tarballFilename,
-      contentType: 'application/x-rpm'
-    };
-  } else if (buildVariant === BuildVariant.Debian) {
-    await tarballDebian(packageInformation, packageInformation.debTemplateDir, tarballFilename);
+  const tarballFile = getTarballFile(buildVariant, packageInformation.metadata.version, packageInformation.metadata.name);
+  const fullTarballFilePath = path.join(outputDir, tarballFile.path);
+  console.info('mongosh: gzipping:', fullTarballFilePath);
 
-    return {
-      path: tarballFilename,
-      contentType: 'application/vnd.debian.binary-package'
-    };
+  switch (buildVariant) {
+    case BuildVariant.Linux:
+      await tarballPosix(packageInformation, fullTarballFilePath);
+      break;
+    case BuildVariant.Redhat:
+      await tarballRedhat(packageInformation, packageInformation.rpmTemplateDir, fullTarballFilePath);
+      break;
+    case BuildVariant.Debian:
+      await tarballDebian(packageInformation, packageInformation.debTemplateDir, fullTarballFilePath);
+      break;
+    case BuildVariant.MacOs:
+    case BuildVariant.Windows:
+      await tarballWindows(packageInformation, fullTarballFilePath);
+      break;
+    default:
+      throw new Error(`Unhandled build variant: ${buildVariant}`);
   }
-  await tarballWindows(packageInformation, tarballFilename);
 
   return {
-    path: tarballFilename,
-    contentType: 'application/zip'
+    ...tarballFile,
+    path: fullTarballFilePath
```

Comment:
I opened a separate PR with the changes to this file in https://github.com/mongodb-js/mongosh/pull/615, because I’ll have to modify it and this patch is not something I’d want to rebase against :D